config/cache/v1alpha1/cache.proto

```diff
@@ -0,0 +1,123 @@
+message CacheSpec {
+  // Resource profile for the cache provider
+  ResourcesProfile resources = 1;
+  // DatasourceRef or a ServiceBindingRef (TODO clarify)
+  DataSourceSpec dataSource = 2;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  DBType dbType = 1;
+  // Additional properties. DB specific
+  map<string,string> connectionProperties = 2;
+  // Reference to the ServiceBinding CR
+  NamespacedRef serviceBindingRef = 3;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message EagerCachingRuleSpec {
+  // Reference to the related Cache CR
+  NamespacedRef  cacheRef = 1;
+  // Resource profile for the DB syncer
+  ResourcesProfile resources = 2;
+  // Name of the table from where the data will be produced. Format could change depending
+  // on the DB: table or schema.table must be at least supported
+  string tableName = 3;
+  // Format of the key for the get(key) operation
+  Key key = 4;
+  // Query columns used to build the entry value
+  Value value = 5;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message LazyCachingRuleSpec {
+  // Reference to the related Cache CR
+  NamespacedRef cacheRef = 1;
+  // The select query needed to fetch values from the DB
+  string query = 2;
+  Value value = 3;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes how the key is build from the query result row
+message Key {
+  // Format of the key for the get(key) operation
+  KeyFormat format = 1;
+  // Separator character in case of plain test key format
+  string keySeparator = 2;
+  // Table columns composing the primary key
+  repeated string keyColumns = 3;
+
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes how the entry value is build from the query result row
+message Value {
+  // Table columns that will be fetched from the DB (select clause)
+  repeated string valueColumns = 1;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a resources profile required for a workload
+message ResourcesProfile {
+  // Memory resource limit for the cache provider
+  k8s.io.apimachinery.pkg.api.resource.Quantity memoryLimits = 1;
+  // Memory resource request for the cache provider
+  k8s.io.apimachinery.pkg.api.resource.Quantity memoryRequests = 2;
+  // CPU resource limit for the cache provider
+  k8s.io.apimachinery.pkg.api.resource.Quantity cpuLimits = 3;
+  // CPU resource request for the cache provider
+  k8s.io.apimachinery.pkg.api.resource.Quantity cpuRequests = 4;
+}
```

Comment:
This still won't represent the standard k8s resource yaml of:

```yaml
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "500m"
```

Instead it will be:

```yaml
resources:
  memoryLimits:
  memoryRequests:
  cpuLimits:
  cputRequests:
```

Can we re-use the .proto k8s uses here?

Reply:
oh right, yeah let me fix it

Reply:
oh no wait, we can't reuse the k8s struct since it's actually a map [see](https://github.com/kubernetes/api/blob/8384a1b0af6a34368da256be0b98035e4f272024/core/v1/types.go#L5286)
an that was my initial implementation.
If we're sure we will just need CPU and mem I can simplify it with a static struct

Reply:
Yeah let's keep it simple. The main thing is that we emulate the k8s yaml, the internals don't matter to the user.